From 20020250a02e3e6c5813a24012bb7454a6195f51 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Stefan=20Br=C3=BCns?= <stefan.bruens@rwth-aachen.de>
Date: Sun, 14 Jul 2019 19:00:49 +0200
Subject: [PATCH] Unified inclusion of exiv2 library [1828925]

Before only the needed include files were used. This is depreacated.
Exiv2 wants only to include the main <exiv2/exiv2.hpp> file
---
 src/hugin1/hugin/AboutDialog.cpp         | 2 +-
 src/hugin1/hugin/PanoOperation.cpp       | 3 +--
 src/hugin1/hugin/RawImport.cpp           | 5 +----
 src/hugin1/hugin/TextureManager.cpp      | 3 +--
 src/hugin1/ptbatcher/FindPanoDialog.cpp  | 3 +--
 src/hugin_base/panodata/Exiv2Helper.cpp  | 5 ++---
 src/hugin_base/panodata/SrcPanoImage.cpp | 4 +---
 7 files changed, 8 insertions(+), 17 deletions(-)

diff --git a/src/hugin1/hugin/AboutDialog.cpp b/src/hugin1/hugin/AboutDialog.cpp
index 5603957..d1291aa 100755
--- a/src/hugin1/hugin/AboutDialog.cpp
+++ b/src/hugin1/hugin/AboutDialog.cpp
@@ -42,7 +42,7 @@ extern "C"
 // workaround for a conflict between exiv2 and wxWidgets/CMake built
 #define HAVE_PID_T 1
 #endif
-#include "exiv2/exiv2.hpp"
+#include <exiv2/exiv2.hpp>
 #include "lensdb/LensDB.h"
 #include "sqlite3.h"
 #include <lcms2.h>
diff --git a/src/hugin1/hugin/PanoOperation.cpp b/src/hugin1/hugin/PanoOperation.cpp
index cf7a6c3..7a3f0d7 100755
--- a/src/hugin1/hugin/PanoOperation.cpp
+++ b/src/hugin1/hugin/PanoOperation.cpp
@@ -36,8 +36,7 @@
 // workaround for a conflict between exiv2 and wxWidgets/CMake built
 #define HAVE_PID_T 1
 #endif
-#include <exiv2/exif.hpp>
-#include <exiv2/image.hpp>
+#include <exiv2/exiv2.hpp>
 #include "base_wx/LensTools.h"
 #include "base_wx/wxLensDB.h"
 #include "hugin/ResetDialog.h"
diff --git a/src/hugin1/hugin/RawImport.cpp b/src/hugin1/hugin/RawImport.cpp
index 2dd4eb9..73bb143 100755
--- a/src/hugin1/hugin/RawImport.cpp
+++ b/src/hugin1/hugin/RawImport.cpp
@@ -35,10 +35,7 @@
 // workaround for a conflict between exiv2 and wxWidgets/CMake built
 #define HAVE_PID_T 1
 #endif
-#include <exiv2/exif.hpp>
-#include <exiv2/image.hpp>
-#include <exiv2/easyaccess.hpp>
-#include <exiv2/xmpsidecar.hpp>
+#include <exiv2/exiv2.hpp>
 #ifdef __WXMSW__
 #include <wx/msw/registry.h>
 #endif
diff --git a/src/hugin1/hugin/TextureManager.cpp b/src/hugin1/hugin/TextureManager.cpp
index 0d2f5da..9a0822a 100755
--- a/src/hugin1/hugin/TextureManager.cpp
+++ b/src/hugin1/hugin/TextureManager.cpp
@@ -65,8 +65,7 @@
 // workaround for a conflict between exiv2 and wxWidgets/CMake built
 #define HAVE_PID_T 1
 #endif
-#include "exiv2/exiv2.hpp"
-#include "exiv2/preview.hpp"
+#include <exiv2/exiv2.hpp>
 
 TextureManager::TextureManager(HuginBase::Panorama *pano, ViewState *view_state_in)
 {
diff --git a/src/hugin1/ptbatcher/FindPanoDialog.cpp b/src/hugin1/ptbatcher/FindPanoDialog.cpp
index 64774fb..96cd762 100755
--- a/src/hugin1/ptbatcher/FindPanoDialog.cpp
+++ b/src/hugin1/ptbatcher/FindPanoDialog.cpp
@@ -36,8 +36,7 @@
 // workaround for a conflict between exiv2 and wxWidgets/CMake built
 #define HAVE_PID_T 1
 #endif
-#include "exiv2/exiv2.hpp"
-#include "exiv2/preview.hpp"
+#include <exiv2/exiv2.hpp>
 #ifdef _WIN32
 #include <commctrl.h>
 #endif
diff --git a/src/hugin_base/panodata/Exiv2Helper.cpp b/src/hugin_base/panodata/Exiv2Helper.cpp
index bc4c12a..5f8bcea 100755
--- a/src/hugin_base/panodata/Exiv2Helper.cpp
+++ b/src/hugin_base/panodata/Exiv2Helper.cpp
@@ -28,8 +28,7 @@
 #include "Exiv2Helper.h"
 #include "hugin_math/hugin_math.h"
 #include "hugin_utils/utils.h"  
-#include "exiv2/easyaccess.hpp"
-#include "exiv2/version.hpp"
+#include <exiv2/exiv2.hpp>
 
 namespace HuginBase
 {
@@ -601,4 +600,4 @@ namespace HuginBase
         };
 
     }; //namespace Exiv2Helper
-}; //namespace HuginBase
\ No newline at end of file
+}; //namespace HuginBase
diff --git a/src/hugin_base/panodata/SrcPanoImage.cpp b/src/hugin_base/panodata/SrcPanoImage.cpp
index cc6a565..8484dd9 100755
--- a/src/hugin_base/panodata/SrcPanoImage.cpp
+++ b/src/hugin_base/panodata/SrcPanoImage.cpp
@@ -40,9 +40,7 @@
 #include <vigra/diff2d.hxx>
 #include <vigra/imageinfo.hxx>
 #include <hugin_utils/utils.h>
-#include <exiv2/exif.hpp>
-#include <exiv2/image.hpp>
-#include <exiv2/easyaccess.hpp>
+#include <exiv2/exiv2.hpp>
 #include <lensdb/LensDB.h>
 #include "Exiv2Helper.h"
 
-- 
2.22.0

